code/modules/mob/mob.dm

```diff
@@ -634,6 +634,13 @@ var/list/slot_equipment_priority = list( \
 	if(!istype(W)) return 0
 
 	for(var/slot in slot_equipment_priority)
+		if(slot == slot_belt) //prioritize placing things in belt rather than swapping the belt slot
```

Comment:
Aaaah the hardcode.

My suggestion would be to give all items a proc like `can_quick_store(var/obj/item/I, var/mob/user)`. Then make it loop through the slots and if the item returns true on `can_quick_store()`, it'll try to insert with a proc like `quick_store(var/obj/item/I, var/mob/user)`